target_chains/ethereum/contracts/contracts/pulse/Pulse.sol

```diff
@@ -85,21 +87,20 @@ abstract contract Pulse is IPulse, PulseState {
         req.requester = msg.sender;
         req.numPriceIds = uint8(priceIds.length);
         req.provider = provider;
+        req.fee = SafeCast.toUint128(msg.value - _state.pythFeeInWei);
```

Comment:
Store the fee in the request so we can credit the provider *who actually fulfilled the request* in `executeCallback`